sdk/dnsresolver/Azure.ResourceManager.DnsResolver/tests/Tests/DnsResolverTests.cs

```diff
@@ -15,17 +15,20 @@ public class DnsResolverTests : DnsResolverTestBase
     {
         private DnsResolverCollection _dnsResolverCollection;
 
-        public DnsResolverTests(bool async) : base(async)
+        public DnsResolverTests(bool async) : base(async)//, RecordedTestMode.Record)
         {
         }
 
-        [SetUp]
         public async Task CreateDnsResolverCollectionAsync()
         {
             var subscription = await Client.GetSubscriptions().GetAsync(TestEnvironment.SubscriptionId);
-            var resourceGroup = await subscription.Value.GetResourceGroups().GetAsync(TestEnvironment.ResourceGroup);
+            var resourceGroup = await CreateResourceGroupAsync();
+            if (Mode == RecordedTestMode.Record || Mode == RecordedTestMode.Playback)
+            {
+                await CreateVirtualNetworkAsync();
+            }
```

Comment:
From the method name, we do not need to do this here - I recommend that we remove this here, and call this in the real test case method below

Reply:
fixed